azurerm failing with Error: Plugin did not respond

Hi there, 

I am receiving this error while running apply on terraform for azure. Below is the stack trace:

      + source_image_reference {
          + offer     = "cis-windows-server-2019-v1-0-0-l1"
          + publisher = "center-for-internet-security-inc"
          + sku       = "cis-ws2019-l1"
          + version   = "latest"
        }

- Installing hashicorp/azurerm v2.90.0...
- Installed hashicorp/azurerm v2.90.0 (signed by HashiCorp)



```
resource "azurerm_managed_disk" "windows_data" {
    count                = var.windows_count
    name                 = "${var.vm_name_prefix}${var.windows_vm_app_code}${format("%d", count.index + 1)}-data-disk"
    location             = var.az_location
    resource_group_name  = var.windows_rsg_name
    storage_account_type = var.managed_disk_storage_type
    create_option        = "Empty"
    disk_size_gb         = var.managed_disk_size_gb
    tags                 = var.tags
}

```

azurerm_windows_virtual_machine.windows_vm[0]: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_windows_virtual_machine.windows_vm[0],
│   on main.tf line 31, in resource "azurerm_windows_virtual_machine" "windows_vm":
│   31: resource "azurerm_windows_virtual_machine" "windows_vm" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_managed_disk.windows_data[0],
│   on main.tf line 77, in resource "azurerm_managed_disk" "windows_data":
│   77: resource "azurerm_managed_disk" "windows_data" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-azurerm_v2.90.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 142 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.expandPlan(...)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/compute/shared_schema.go:300
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.resourceWindowsVirtualMachineCreate(0x0, {0x60eebe0, 0xc001503500})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/compute/windows_virtual_machine_resource.go:396 +0x2d65
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc001c902d0, {0x70be178, 0xc001b1ca80}, 0x2, {0x60eebe0, 0xc001503500})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:329 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0008130a0, {0x70be178, 0xc001b1ca80}, 0xc001b348f0, 0xc001b30880, {0x60eebe0, 0xc001503500})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000415c20, {0x70be178, 0xc001b1ca80}, 0xc000113540)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000290500, {0x70be220, 0xc001b0f0b0}, 0xc000c96cb0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x66242e0, 0xc000290500}, {0x70be220, 0xc001b0f0b0}, 0xc001b2c000, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000241340, {0x710c0c8, 0xc00019b380}, 0xc001b20fc0, 0xc0007a5500, 0xb61a4c0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000241340, {0x710c0c8, 0xc00019b380}, 0xc001b20fc0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.90.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

ERRO[0041] 1 error occurred:
	* exit status 1

please let me know if this is something we have to fix in our setup or this something plugin related

--- a/internal/services/compute/shared_schema.go
+++ b/internal/services/compute/shared_schema.go
@@ -96,7 +96,7 @@ func flattenAdditionalUnattendContent(input *[]compute.AdditionalUnattendContent
 }
 
 func bootDiagnosticsSchema() *pluginsdk.Schema {
-	//lintignore:XS003
+	// lintignore:XS003
 	return &pluginsdk.Schema{
 		Type:     pluginsdk.TypeList,
 		Optional: true,
@@ -293,7 +293,7 @@ func planSchema() *pluginsdk.Schema {
 }
 
 func expandPlan(input []interface{}) *compute.Plan {
-	if len(input) == 0 {
+	if input == nil || len(input) == 0 {
 		return nil
 	}
 

--- a/internal/services/compute/shared_schema.go
+++ b/internal/services/compute/shared_schema.go
@@ -293,7 +293,7 @@ func planSchema() *pluginsdk.Schema {
 }
 
 func expandPlan(input []interface{}) *compute.Plan {
-	if input == nil || len(input) == 0 {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 

